simulator/infra/macro.h

```diff
@@ -173,6 +173,13 @@ static constexpr T ones_ls(const T& value, size_t shamt)
     return ~(~value << shamt);
 }
 
+/*Circular shift for the value with XLEN bits*/
+template<typename T>
+static constexpr T circ_s(const T& value, size_t shamt, size_t XLEN)
```

Comment:
Could you please clarify whether it is a left or right shift?
Additionally, XLEN may be deduced here as `bitwidth<T>` without additional parameter.